lib/internal/process/promises.js

```diff
@@ -26,6 +33,7 @@ function setupPromises(scheduleMicrotasks) {
     if (hasBeenNotified !== undefined) {
       hasBeenNotifiedProperty.delete(promise);
       if (hasBeenNotified === true) {
+        promiseInternals.untrackPromise(promise);
```

Comment:
I don't think such a check is really required here (dealing with cases of 1000s of unhandled rejections).